### PR TITLE
[4.0] Changing title to tooltip for template preview

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
As title says. 

### Testing Instructions
Load Template styles manager and switch Options => Preview Module Positions on and off.
Same for Templates Manager

Hover the Preview icon

### Actual result BEFORE applying this Pull Request
Screen Shot 2021-04-25 at 08 49 11

Screen Shot 2021-04-25 at 08 50 34

Screen Shot 2021-04-25 at 08 51 43

### Expected result AFTER applying this Pull Request

Screen Shot 2021-04-25 at 08 53 26

Screen Shot 2021-04-25 at 09 06 41

Screen Shot 2021-04-25 at 08 53 58

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 352b4c32698684c3a30c9539214ba7e2939e0499<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

**Comment by brianteeman** [comment]
This is not correct. Please test with a screen reader.

**Comment by infograf768** [comment]
1. I have no screenreader except Mac VoiceOver. 
Not sure what would be the correct preferences but using default I tested it on Firefox. It does not read the title either.
2. We have other occurrences in admin with similar code and VoiceOver does not read the tips either.

If it is no correct, then propose something that works.

**Comment by infograf768** [comment]
Explaining more.
When we have a link I can add an `aria-labelledby` and it looks like working
```
<a href="<?php echo Route::link($client, 'index.php?tp=1&templateStyle=' . (int) $item->id); ?>" target="_blank" class="jgrid" aria-labelledby="preview-<?php echo (int) $item->id; ?>">
											<span class="visually-hidden"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?></span>
										</a>
										<div role="tooltip" id="preview-<?php echo (int) $item->id; ?>"><?php echo Text::sprintf('JBROWSERTARGET_NEW_TITLE', $item->title); ?></div>
```

But I have no idea how to do when we have no link. I.e. when no preview.
```
<?php else : ?>
										<span class="icon-eye-slash" aria-hidden="true"></span>
										<div role="tooltip" id="nopreview-<?php echo (int) $item->id; ?>"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_NO_PREVIEW'); ?></div>
										<span class="visually-hidden"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_NO_PREVIEW'); ?></span>
									<?php endif; ?>
```

**Comment by brianteeman** [comment]
What is the purpose of `<span class="visually-hidden"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?></span>`

**Comment by infograf768** [comment]
> What is the purpose of `<span class="visually-hidden"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?></span>`

I did not touch that, It was there before as you can see in the diff.

**Comment by infograf768** [comment]
So normally, the PR is now OK by adding `aria-labelledby`.

anything else can be modified if necessary in another PR.

Now preparing another PR to use `aria-labelledby` for associations badges.

**Comment by brianteeman** [comment]
> I did not touch that, It was there before as you can see in the diff.

Maybe not but your changes have an impact on the existing code

**Comment by infograf768** [comment]
no idea what you mean. Want me to also close that one on a useless comment?

**Comment by brianteeman** [comment]
OK let me try again with a picture after applying the PR

image

So the point is that you should have touched that line as it is now redundant

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on edf8bbf4d3db438aaa527da4289480a998074ba0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on edf8bbf4d3db438aaa527da4289480a998074ba0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

**Comment by infograf768** [comment]
@brianteeman 
Do you mean that 
`class="visually-hidden">` should be deleted?

**Comment by brianteeman** [comment]
No I mean that the entire line should be deleted as there is no need to have it twice

image

(ignore the highlight in the image)

**Comment by brianteeman** [comment]
(PS hope you are well - not seen you online for longer than usual)

**Comment by richard67** [comment]
@sandramay0905 @RickR2H Could you test this PR again? It has received changes. Thanks in advance.

**Comment by infograf768** [comment]
Deleted redundant span.

Afterthought:
Using `JBROWSERTARGET_NEW_TITLE="Open %s in new window"` may be not specific enough.

What do you think about using a new string:
`COM_TEMPLATES_TEMPLATE_NEW_PREVIEW="Preview %s in new window"`

**Comment by brianteeman** [comment]
Open or Preview - both are fine - whichever you prefer is fine by me

**Comment by infograf768** [comment]
Used new string

Screen Shot 2021-05-12 at 10 34 06
Screen Shot 2021-05-12 at 10 33 44

can be tested now

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 5442371c5fb1d239e59b891c1562255128815fac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

File: administrator/components/com_templates/tmpl/styles/default.php [hunk 1]
```diff
@@ -84,13 +84,12 @@
...
-										<a href="<?php echo Route::link($client, 'index.php?tp=1&templateStyle=' . (int) $item->id); ?>" target="_blank" class="jgrid"
-											title="<?php echo Text::sprintf('JBROWSERTARGET_NEW_TITLE', $item->title); ?>">
-											<span class="visually-hidden"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?></span>
-										</a>
+										<a href="<?php echo Route::link($client, 'index.php?tp=1&templateStyle=' . (int) $item->id); ?>" target="_blank" class="jgrid" aria-labelledby="preview-<?php echo (int) $item->id; ?>"></a>
+										<div role="tooltip" id="preview-<?php echo (int) $item->id; ?>"><?php echo Text::sprintf('COM_TEMPLATES_TEMPLATE_NEW_PREVIEW', $item->title); ?></div>
 									<?php else : ?>
-										<span class="icon-eye-slash" aria-hidden="true" title="<?php echo Text::_('COM_TEMPLATES_TEMPLATE_NO_PREVIEW'); ?>"></span>
-										<span class="visually-hidden"><?php echo Text::_('COM_TEMPLATES_TEMPLATE_NO_PREVIEW'); ?></span>
+										<a href="#" aria-labelledby="nopreview-<?php echo (int) $item->id; ?>">
```
Comment (Quy) [inline administrator/components/com_templates/tmpl/styles/default.php]: It is missing `</a>`. Plus it gives the impression that the button is clickable when it is not.

Comment (infograf768) [reply administrator/components/com_templates/tmpl/styles/default.php]: indeed. We do not need the `<a` at all.

**Comment by infograf768** [comment]
Thanks @Quy 
I added the `labelledby` directly in the span. I guess that would be OK as it also takes care of the cursor.

tip

**Comment by infograf768** [comment]
@sandramay0905 @RickR2H Could you test this PR again?  Modified some aspects. test specially when no preview in template styles.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on b57d2454c04777c67d8806d029d8bd8bd02726d9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

File: administrator/components/com_templates/tmpl/templates/default.php [hunk 2]
```diff
@@ -71,11 +71,10 @@
 									<div>
 										<?php if ($this->preview) : ?>
 											<?php $client = (int) $item->client_id === 1 ? 'administrator' : 'site'; ?>
-											<a href="<?php echo Route::link($client, 'index.php?tp=1&template=' . $item->element); ?>"
-												target="_blank"
-												title="<?php echo Text::sprintf('JBROWSERTARGET_NEW_TITLE', Text::_($item->name)); ?>">
-											<?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?>
+											<a href="<?php echo Route::link($client, 'index.php?tp=1&template=' . $item->element); ?> "target="_blank" aria-labelledby="preview-<?php echo $item->extension_id; ?>">
```
Comment (Quy) [inline administrator/components/com_templates/tmpl/templates/default.php]:
```suggestion
											<a href="<?php echo Route::link($client, 'index.php?tp=1&template=' . $item->element); ?>" target="_blank" aria-labelledby="preview-<?php echo $item->extension_id; ?>">
```

Comment (infograf768) [reply administrator/components/com_templates/tmpl/templates/default.php]: done
does not change test

**Comment by ghost** [comment]
@RickR2H can you please retest?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 490cfb061ecaf29f18790a22124bd969ab4f470f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33292">issues.joomla.org/tracker/joomla-cms/33292</a>.</sub>

**Comment by chmst** [comment]
Thanks!
